src/controllers/users.controller.js

```diff
@@ -0,0 +1,86 @@
+const { usersService } = require('./../services');
+const { userSchema } = require('./../schemas');
+
+const getAll = (req, res) => {
+  const users = usersService.getAll();
+
+  res.json(users);
+};
+
+const getOne = (req, res) => {
+  const { id } = req.params;
+
+  const user = usersService.getById(+id);
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.json(user);
+};
+
+const create = (req, res) => {
+  const { error, value } = userSchema.validate(req.body, {
+    stripUnknown: true,
+  });
+
+  if (error) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = usersService.create(value);
+
+  res.status(201).json(newUser);
+};
+
+const update = (req, res) => {
+  const { id } = req.params;
+  const { error, value } = userSchema.validate(req.body, {
+    stripUnknown: true,
+  });
+
+  if (error) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const isExist = !!usersService.getById(+id);
+
+  if (!isExist) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const updatedUser = usersService.update(+id, value);
```

Comment:
Handle potential errors from the `usersService.update` method, such as database errors or invalid updates, and respond with an appropriate status code and message.